hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/dml/TestInsertTable.scala

```diff
@@ -294,6 +294,37 @@ class TestInsertTable extends HoodieSparkSqlTestBase {
     }
   }
 
+  test("Test Insert Into Table with partition field not in the end") {
```

Comment:
can't reproduct use this case in master branch with spark32, this test will success in my local and create normal partions. did I ignore some details?
<img width="268" alt="image" src="https://github.com/apache/hudi/assets/20125927/d672afca-085a-4567-b60b-dfd36acf18f3">
<img width="957" alt="image" src="https://github.com/apache/hudi/assets/20125927/5d97767b-e392-426f-af31-01d5b8c33446">